web/packages/teleterm/src/ui/DocumentCluster/useUserPreferences.ts

```diff
@@ -138,7 +124,8 @@ export function useUserPreferences(clusterUri: ClusterUri): {
         const [prefs, error] = await runInitialFetchAttempt();
         if (!error) {
           updateUnifiedResourcePreferencesStateAndWorkspace(
-            prefs?.unifiedResourcePreferences
+            prefs?.unifiedResourcePreferences ||
+              getDefaultUnifiedResourcePreferences()
```

Comment:
I had to add a default here because somewhere in the tests we were getting undefined. I couldn't trace it in the mock contexts but let me know where I'm missing it, as it'd be nice to not have here

Reply:
Some tests were failing because the hook calls `tshd.getUserPreferences` which returns an empty object in tests.
I believe we should just update the mock so it returns a full object, just as a regular call.
In `web/packages/teleterm/src/services/tshd/fixtures/mocks.ts`:
```ts
  getUserPreferences = () =>
    new MockedUnaryCall({
      userPreferences: {
        unifiedResourcePreferences: getDefaultUnifiedResourcePreferences(),
        clusterPreferences: { pinnedResources: { resourceIds: [] } },
      },
    });
```

We should also be able to remove the optional chaining from `prefs` (here and in the other places in the hook), the response is always defined.

Reply:
I forgot to say it in the previous message, but with this mock added, we can get rid of `getDefaultUnifiedResourcePreferences()` from the linked line.